### PR TITLE
sof:suecreek:add the initial Sue Creek support

### PR DESCRIPTION
**Pull request by zhigang-wu**
1. fix typo erro to pass codecheck.
2. update the script
3. add the sue-ipc.c file

Signed-off-by: Wu Zhigang <zhigang.wu@linux.intel.com>

### CODE REVIEW

File: src/ipc/apl-ipc.c [hunk 1]
```diff
@@ -215,8 +215,8 @@ int platform_ipc_init(struct ipc *ipc)
 	iipc->pm_prepare_D3 = 0;
 
 	/* configure interrupt */
-	interrupt_register(PLATFORM_IPC_INTERUPT, irq_handler, NULL);
-	interrupt_enable(PLATFORM_IPC_INTERUPT);
+	interrupt_register(PLATFORM_IPC_INTERRUPT, irq_handler, NULL);
```
Comment (lgirdwood) [inline src/ipc/apl-ipc.c]: This spelling fix should be a separate patch for master branch.

File: src/ipc/handler.c [hunk 2]
```diff
@@ -74,6 +74,7 @@ static inline struct sof_ipc_hdr *mailbox_validate(void)
 {
 	struct sof_ipc_hdr *hdr = _ipc->comp_data;
 
+#if !defined CONFIG_SUECREEK
 	/* read component values from the inbox */
 	mailbox_hostbox_read(hdr, 0, sizeof(*hdr));
```
Comment (lgirdwood) [inline src/ipc/handler.c]: ok, so lets make mailbox_read on SUE read the SPI buffer, that way we wont need the #ifdef

**Comment by lgirdwood** [comment]
@zhigang-wu I've updated code to build now. Can you send spelling fix to master.

**Comment by zhigang-wu** [comment]
@lgirdwood 
Seems you merged all of them already, I will close this PR.

File: src/ipc/apl-ipc.c (hunk 1 above)
Comment (zhigang-wu) [reply src/ipc/apl-ipc.c]: ok. will do this.
